src/main/scala/io/github/kiemlicz/shelm/ChartDownloader.scala

```diff
@@ -33,18 +51,55 @@ object ChartDownloader {
           )
         else
           downloadDir / topDirs.head
-      case ChartLocation.AddedRepository(ChartName(name), ChartRepositoryName(repoName), chartVersion) =>
-        val options = s"$repoName/$name -d $downloadDir${chartVersion.map(v => s" --version $v").getOrElse("")} --untar"
-        IO.delete(downloadDir)
-        pullChart(options, sbtLogger)
-        downloadDir / name
-      case ChartLocation.RemoteRepository(ChartName(name), uri, settings, chartVersion) =>
-        val authOpts = HelmPlugin.chartRepositoryCommandFlags(settings)
-        val allOptions = s"--repo $uri $name $authOpts -d $downloadDir${chartVersion.map(v => s" --version $v").getOrElse("")} --untar"
-        IO.delete(downloadDir)
-        pullChart(allOptions, sbtLogger)
-        downloadDir / name
+      case ChartLocation.AddedRepository(chartName, ChartRepositoryName(repoName), chartVersion) =>
+        pullFromCacheOrRemote(
+          chartName, chartVersion, () => {
+            val options = s"$repoName/${chartName.name} -d $downloadDir${
+              chartVersion.map(v => s" --version $v").getOrElse("")
+            } --untar"
+            IO.delete(downloadDir)
+            pullChart(options, sbtLogger)
+          }
+        )
+      case ChartLocation.RemoteRepository(chartName, uri, settings, chartVersion) =>
+        pullFromCacheOrRemote(
+          chartName, chartVersion, () => {
+            val authOpts = HelmPlugin.chartRepositoryCommandFlags(settings)
+            val allOptions = s"--repo $uri ${chartName.name} $authOpts -d $downloadDir${
+              chartVersion.map(v => s" --version $v").getOrElse("")
+            } --untar"
+            IO.delete(downloadDir)
+            pullChart(allOptions, sbtLogger)
+          }
+        )
+    }
+  }
+
+  def calcCacheChartDir(
```

Comment:
Scala uses Java-ish approach of giving everything a full name, please don't abbreviate `calc` here